Dynamic Transforms

Spartwo opened the pull request:
Adds ability for command pods to shift their frame of reference on demand alongside supporting changes.
- Missiles now point forward based on thrust direction
- Command pods come with a "dynamic" control point that shifts when needed
- Added support for rcs to thrust calculations again
- Added aformentioned transform movement
- Shifted minimum value for firework burst spacing above 0
- Added initial supports into rockets for future rework of logic

Spartwo committed:
push

Spartwo committed:
debug performance

make debug lines check for debug mode before pre-processing

Spartwo committed:
Decoupler fix

- Fix initial endcap seperation function to specify type and ignore warheads

Halbann committed:
Create dynamic control points at runtime

Halbann committed:
Fixes

Fix propulsion line, Make RCS list optional, Fix docking port separation, Separator spelling change

Halbann committed:
remove empty line

Halbann committed:
Simplified FindDecoupler params

Halbann committed:
FIndDecoupler Fix

Spartwo committed:
I'm learnding

default values(thanks hatbat)

Halbann committed:
Update Source/KerbalCombatSystems/MissileGuidance.cs

Co-authored-by: Spartwo <38593593+Spartwo@users.noreply.github.com>

Halbann committed:
Update Source/KerbalCombatSystems/MissileGuidance.cs

Co-authored-by: Spartwo <38593593+Spartwo@users.noreply.github.com>

Halbann committed:
Update Source/KerbalCombatSystems/Utils.cs

Co-authored-by: Spartwo <38593593+Spartwo@users.noreply.github.com>

Halbann committed:
Update Source/KerbalCombatSystems/Utils.cs

Co-authored-by: Spartwo <38593593+Spartwo@users.noreply.github.com>

Halbann committed:
Update Source/KerbalCombatSystems/Utils.cs

Co-authored-by: Spartwo <38593593+Spartwo@users.noreply.github.com>

Spartwo committed:
Revert "I'm learnding"

This reverts commit 67bc8d24d58f70f35d5d4c3429743a088147006c.

Halbann committed:
Revert "Utils changes"

This reverts commit 75c3eca2218df7c9fb3cd290b08fddbab595a25d.

Halbann committed:
Utils changes

Spartwo committed:
Merge pull request #4 from Halbann/suggestions

Suggestions for Dynamic Transforms

Spartwo committed:
Dynamic Precheck

Check for already existing dynamic control point

Spartwo committed:
origin removal

removed no longer required origin variable in firevector call

Halbann commented:
WIll merge after the accuracy problems we talked about (if they are actually to do with the thrust vector) are figured out.

Spartwo committed:
MeanThrust Fixes

- Fixed nonrecognition of rcs only missiles
- Replaced main engine as was exclusionary to multistaging, rcs, and clusters of engines
- Modified start sequence

Spartwo committed:
Update MissileGuidance.cs

Spartwo committed:
RCS TWR calcs

Improved fire vector calculations accounting for rcs and other edge cases
